tests/unit/plugins/modules/test_ec2_ami_info.py

```diff
@@ -0,0 +1,253 @@
+# (c) 2022 Red Hat Inc.
+
+# This file is part of Ansible
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from ansible.module_utils.common.dict_transformations import camel_dict_to_snake_dict
+from unittest.mock import MagicMock, patch, ANY, call
+import botocore.exceptions
+import pytest
+
+from ansible_collections.amazon.aws.plugins.modules import ec2_ami_info
+
+module_name = "ansible_collections.amazon.aws.plugins.modules.ec2_ami_info"
+
+
+@pytest.mark.parametrize("executable_users,filters,image_ids,owners,expected", [
+    ([], {}, [], [], {}),
+    ([], {}, ['ami-1234567890'], [], {'ImageIds': ['ami-1234567890']}),
+    ([], {}, [], ['1234567890'], {'Filters': [{'Name': 'owner-id', 'Values': ['1234567890']}]}),
+    ([], {'owner-alias': 'test_ami_owner'}, [], ['1234567890'], {'Filters': [{'Name': 'owner-alias',
+     'Values': ['test_ami_owner']}, {'Name': 'owner-id', 'Values': ['1234567890']}]}),
+    ([], {'is-public': True}, [], [], {'Filters': [{'Name': 'is-public', 'Values': ['true']}]}),
+    ([], {}, [], ['self'], {'Owners': ['self']})])
+def test_build_request_args(executable_users, filters, image_ids, owners, expected):
+    assert ec2_ami_info.build_request_args(
+        executable_users, filters, image_ids, owners) == expected
+
+
+def test_get_images():
```

Comment:
you should add a test case for the failure

Reply:
yes, already added https://github.com/ansible-collections/amazon.aws/pull/1252/files#diff-6192636df21722bd5081088ae5e21132bb06367a3b687c2ea8e64553759a19acR237-R253